app/scripts/models/auth_brokers/oauth.js

```diff
@@ -221,7 +221,13 @@ var OAuthAuthenticationBroker = BaseAuthenticationBroker.extend({
   },
 
   afterResetPasswordConfirmationPoll (account) {
-    return this.finishOAuthSignInFlow(account);
+    return Promise.resolve().then(() => {
```

Comment:
the distinction between oauth and oauth-redirect has outlived its usefulness, we should merge the two. *Not* for this PR!